site/content/docs/development/architecture.md

```diff
@@ -0,0 +1,142 @@
+* **Source**: The source reference for the input to the Supply Chain or Delivery Blueprints,
+see [Workload](reference.md/#workload) and [Deliverable](reference.md/#deliverable)
+
+![Owner](../img/owner.jpg)
+
+| Owner      | Blueprint |
+| ----------- | ----------- |
+| Workload | ClusterSupplyChain |
+| Deliverable | ClusterDelivery |
+
+## Theory of operation
+
+Given an owner that matches a blueprint, cartographer reconciles the resources referenced by the blueprint.
+The resources are only created when the inputs are satisfied, and a resource is only updated when it's inputs change.
+This results in a system where a new intrinsic result from one resource can cause other resources to change.
+
+Although Cartographer is not a 'runner of things', a resource can be something as simple as a Job or a CI pipeline.
+However, one advantage of Cartographer's design, is that a resource can also be untriggered. Imagine a Build resource 
+that discovers new base OCI images. If it rebuilds your image, then cartographer will see this new image and update 
+other linked resources.
+
+![Generic Blueprint](../img/generic.jpg)
+<!-- https://miro.com/app/board/uXjVOeb8u5o=/ -->
+
+When Cartographer reconciles an owner, each resource in the matching blueprint is reconciled:
+
+1. **Generate Inputs**: Using the **blueprint resource's** `inputs` as a reference, select outputs from previously applied **Kubernetes resources**
+2. **Generate Params**: Using the [Parameter Hierarchy](architecture.md#parameter-hierarchy), generate parameter values   
+3. **Generate and apply resource spec**: Apply the result of interpolating `spec.template` (or `spec.ytt`), **inputs**, **params** and the **owner spec**. 
+4. **Retrieve Output**: Store the output from the applied resource. The output to use is specified in the **template output path**.  
+
+![Realize](../img/realize.jpg)
+
+### Complete Supply Chain and Delivery with GitOps
+
+![Gitops](../img/gitops.jpg)
+
+## Blueprint Details
+
+### ClusterSupplyChain
+A ClusterSupplyChain blueprint continuously integrates and builds your app.
+
+![ClusterSupplyChain](../img/supplychain.jpg)
+
+### ClusterDelivery
+A ClusterDelivery blueprint continuously deploys and validates images to a cluster. A ClusterDelivery has the ability to lock 
+(and unlock) templates which pauses the continuous deploy. 
+
+<!--- @TODO MORE ON LOCKING -->
+
+![ClusterDelivery](../img/delivery.jpg)
+
+### Selectors
+An owner's labels will determine which blueprint will select for it. The controller will do a "best match" on a blueprint's 
+`spec.selector` with an owner's labels.
+
+A "best match" follows the rules:
+1. If all labels are fully contained in the selector, reconcile the owner with that blueprint
+2. If more than one blueprint has all the labels that the owner has, pick the most identical to the owner
+3. If multiple blueprints match the owner labels, reconcile with the blueprint with the most label matches
```

Comment:
question:
I'm a bit confused by these rules. In rule `2`, shouldn't it be the inverse- if an owner has all the labels of more than one blueprint? Also, what does it mean "the most identical to the owner"- does that mean most label matches?

Also I'm not sure what the difference between `2` and `3` is.

Reply:
2. If not all labels match, we choose the blueprint with the most matched labels.
3. If two blueprints match all labels, the blueprint with the more concise match (less non-matching labels) is selected

blueprint a is selected for 

workload
a
b


blueprint a
a
b

blueprint b
a
b
c